tests/test_creation.py

```diff
@@ -0,0 +1,315 @@
+    assert repr(poly_cw) == "POLYGON ((2 0, 2 2, 0 2, 0 0, 2 0))"
+
+
+def test_polygon_normalize_holes() -> None:
+    poly_hole_ccw = spherely.polygon(
+        shell=[(0, 0), (2, 0), (2, 2), (0, 2)],
+        holes=[[(0.5, 0.5), (1.5, 0.5), (1.5, 1.5), (0.5, 1.5)]],
+    )
+
+    # CCW polygon hole vertices reordered
+    # TODO: better to test actual coordinate values when implemented
+    assert (
+        repr(poly_hole_ccw)
+        == "POLYGON ((0 0, 2 0, 2 2, 0 2, 0 0), (0.5 1.5, 1.5 1.5, 1.5 0.5, 0.5 0.5, 0.5 1.5))"
+    )
+
+
+def test_polygon_empty() -> None:
+    poly = spherely.polygon()
+    assert repr(poly).startswith("POLYGON EMPTY")
+
+    poly = spherely.polygon(None)
+    assert repr(poly).startswith("POLYGON EMPTY")
+
+    poly = spherely.polygon([])
+    assert repr(poly).startswith("POLYGON EMPTY")
+
+    with pytest.raises(ValueError, match="with empty component"):
+        spherely.polygon(
+            [spherely.point(5, 50), spherely.point(6, 50), spherely.point()]
+        )
+
+
+def test_polygon_invalid_geography() -> None:
+    shell_points = [
+        spherely.point(0, 0),
+        spherely.point(2, 0),
+        spherely.point(2, 2),
+        spherely.point(0, 2),
+    ]
+    hole_points = [
+        spherely.point(0.5, 0.5),
+        spherely.point(1.5, 0.5),
+        spherely.point(1.5, 1.5),
+        spherely.point(0.5, 1.5),
+    ]
+    line = spherely.linestring([(3, 0), (3, 1)])
+
+    with pytest.raises(
+        TypeError,
+        match=r"invalid Geography type \(expected POINT, found LINESTRING\)",
+    ):
+        spherely.polygon(shell_points + [line])
+
+    with pytest.raises(
+        TypeError,
+        match=r"invalid Geography type \(expected POINT, found LINESTRING\)",
+    ):
+        spherely.polygon(shell=shell_points, holes=[hole_points + [line]])
+
+
+def test_multipolygons() -> None:
+    poly1 = spherely.polygon([(0, 0), (2, 0), (2, 2), (0, 2)])
+    poly2 = spherely.polygon(
+        shell=[(4, 0), (6, 0), (6, 2), (4, 2)],
+        holes=[[(4.5, 0.5), (5.5, 0.5), (5.5, 1.5), (4.5, 1.5)]],
+    )
+    print(spherely.GeographyType(spherely.get_type_id(poly2)))
+
+    multipoly = spherely.multipolygon([poly1, poly2])
+    print(multipoly)
+
+    assert multipoly.dimensions == 2
+    assert multipoly.nshape == 1
+    assert repr(multipoly).startswith("MULTIPOLYGON (((0 0")
+
+
+def test_multipolygon_invalid_geography() -> None:
+    poly = spherely.polygon([(0, 0), (2, 0), (2, 2), (0, 2)])
+    line = spherely.linestring([(3, 0), (3, 1)])
+
+    with pytest.raises(
+        TypeError,
+        match=r"invalid Geography type \(expected POLYGON, found LINESTRING\)",
+    ):
+        spherely.multipolygon([poly, line])
+
+
+def test_collection() -> None:
+    objs = [
+        spherely.point(0, 0),
+        spherely.linestring([(0, 0), (1, 1)]),
+        spherely.polygon([(0, 0), (1, 0), (1, 1)]),
+    ]
+
+    coll = spherely.geography_collection(objs)
+
+    assert coll.dimensions == -1
```

Comment:
This doesn't match with the docstring, stating that for a collection it is the max dimension of the individual features

Reply:
Opened #57.